crates/driver/src/infra/persistence/mod.rs

```diff
@@ -53,11 +55,13 @@ impl Persistence {
 
     /// Saves the given auction with liquidity with fire and forget mentality
     /// (non-blocking operation)
-    pub fn archive_auction(&self, auction_id: Id, body: &str) {
+    pub fn archive_auction(&self, auction_id: Id, body: impl Serialize) {
         let Some(uploader) = self.s3.clone() else {
             return;
         };
-        let body = body.to_string();
+        let Ok(body) = to_value(body) else {
```

Comment:
The reason why I don't pass it directly to the `uploader.upload()` is because we spawn a task, so it would need to live static, and for that we would need to make dto::Auction clonable :(

Reply:
Let's not `expect` for things that are not on the startup path (easy to get crash-loops if this method is ever called by something that doesn't derive Serialize), but I agree we should error log here.